src/hooks/screen/useMessageHook.ts

```diff
@@ -0,0 +1,132 @@
+  const {selectedChannel} = useChatUtilsHook();
+  const [replyPreview, dispatch] = (useContext(Context) as unknown as any).chat;
+  const {anon_user_info_emoji_name} = selectedChannel?.rawJson?.channel || {};
+  const bubblePosition = useSharedValue(0);
+  const pulseAnimation = useSharedValue(1);
+
+  const setReplyPreview = useCallback((messageItem) => {
+    setReplyTarget(messageItem, dispatch);
+  }, []);
+
+  const clearReplyPreview = useCallback(() => {
+    setReplyTarget(null, dispatch);
+  }, []);
+
+  const deleteMessage = async (messageId: string, type: 'ANONYMOUS' | 'SIGNED', iteration = 0) => {
+    if (!localDb) return;
+
+    if (iteration > 5) {
+      SimpleToast.show("Can't delete message, please check your connection");
+      return;
+    }
+
+    try {
+      ChatSchema.updateDeletedChatType(localDb, messageId);
+      refresh('chat');
+
+      const {replyTarget} = replyPreview;
+      if (replyTarget && replyTarget?.id === messageId) {
+        const newReplyPreview = {...replyTarget};
+        newReplyPreview.message = DELETED_MESSAGE_TEXT;
+        newReplyPreview.message_type = 'deleted';
+        setReplyPreview(newReplyPreview);
+      }
+
+      if (type === 'ANONYMOUS') {
+        await AnonymousMessageRepo.deleteMessage(messageId);
+      } else {
+        await SignedMessageRepo.deleteMessage(messageId);
+      }
+    } catch (error) {
+      console.log('error on delete message:', error);
+      setTimeout(() => {
+        deleteMessage(messageId, type, iteration + 1).catch((e) => console.log(e));
+      }, 1000);
+    }
+  };
+
+  const getUserName = (item): string => {
+    if (item?.user?.username !== 'AnonymousUser') {
+      return item?.user?.username;
+    }
+    return `Anonymous ${anon_user_info_emoji_name}`;
+  };
+
+  const onContextMenuPressed = (
+    e: NativeSyntheticEvent<ContextMenuOnPressNativeEvent>,
+    data,
+    type: 'ANONYMOUS' | 'SIGNED'
+  ) => {
+    const event = e.nativeEvent;
+    data = data?.message?.user ? data?.message : data;
+
+    if (event.index === 0) {
+      const messageItem: ReplyMessage = {
+        id: data?.id ?? data?.message?.id,
+        user: {username: getUserName(data)},
+        message: data?.text ?? data?.message,
+        message_type: data?.message_type ?? data?.message?.message_type,
+        updated_at: data?.updated_at ?? data?.message?.updated_at,
+        chatType: type
+      };
+      setReplyPreview(messageItem);
+    }
+    if (event.index === 1) {
+      const isDeleted = data?.message_type === 'deleted';
+      const message = data?.text ?? data?.message;
+      const textToCopy = isDeleted ? '' : message;
+      ShareUtils.copyTextToClipboard(textToCopy);
+    }
+    if (event.index === 2) {
+      Alert.alert('Delete Message?', '', [
+        {text: 'Cancel', style: 'cancel'},
+        {text: 'Confirm', style: 'destructive', onPress: () => deleteMessage(data?.id, type)}
+      ]);
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `onContextMenuPressed` function is quite large and does multiple things. It would be better to split it into smaller functions, each with a single responsibility. This will make the code easier to read and maintain.

```diff
  const onContextMenuPressed = (
    e: NativeSyntheticEvent<ContextMenuOnPressNativeEvent>,
    data,
    type: 'ANONYMOUS' | 'SIGNED'
  ) => {
    const event = e.nativeEvent;
    data = data?.message?.user ? data?.message : data;

    if (event.index === 0) {
      const messageItem: ReplyMessage = {
        id: data?.id ?? data?.message?.id,
        user: {username: getUserName(data)},
        message: data?.text ?? data?.message,
        message_type: data?.message_type ?? data?.message?.message_type,
        updated_at: data?.updated_at ?? data?.message?.updated_at,
        chatType: type
      };
      setReplyPreview(messageItem);
    }
    if (event.index === 1) {
      const isDeleted = data?.message_type === 'deleted';
      const message = data?.text ?? data?.message;
      const textToCopy = isDeleted ? '' : message;
      ShareUtils.copyTextToClipboard(textToCopy);
    }
    if (event.index === 2) {
      Alert.alert('Delete Message?', '', [
        {text: 'Cancel', style: 'cancel'},
        {text: 'Confirm', style: 'destructive', onPress: () => deleteMessage(data?.id, type)}
      ]);
    }
  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->